external-crates/move/crates/move-disassembler/src/disassembler.rs

```diff
@@ -566,30 +564,15 @@ impl<'a> Disassembler<'a> {
         buffer: &mut impl Write,
         function_source_map: &FunctionSourceMap,
         locals_idx: SignatureIndex,
-        parameter_len: usize,
     ) -> Result<()> {
         if !self.options.print_locals {
             return Ok(());
         }
 
-        if function_source_map.locals.len() <= parameter_len {
-            return Ok(());
-        }
-
         let signature = self.source_mapper.bytecode.signature_at(locals_idx);
-        for (local_idx, (name, _)) in function_source_map
-            .locals
-            .iter()
-            .skip(parameter_len)
-            .enumerate()
-        {
+        for (local_idx, (name, _)) in function_source_map.locals.iter().enumerate() {
             any_write!(buffer, "L{local_idx}:\t{name}: ")?;
-            self.disassemble_type_for_local(
-                buffer,
-                function_source_map,
-                parameter_len + local_idx,
-                signature,
-            )?;
+            self.disassemble_type_for_local(buffer, function_source_map, local_idx, signature)?;
```

Comment:
I don't think this is quite right. There's a slight mismatch here due to an underlying problem/mismatch between source maps and compiled modules:
* In the bytecode all parameters are laid out in locals (so the first `parameter_len` local indices in the bytecode are the parameters).
* In the function source map, we add parameters in to the `parameters` field, but not the locals. 

I see two possibilities here:
1. Pass `local_idx + parameter_len` here (since this is the index into the bytecode, so we want to skip the parameters here).
2. Update the source map generation to also add the parameters to the `locals` (and in this case the `parameters` field in the function source map should be removed). 

I prefer option (2) but that's a lot more work than just updating the index here, so if you need this to unblock you updating the index just here is probably fine.

Reply:
I don't think this is going to work as in `disassemble_locals` we actually iterate over locals indexes from the source map:
```move
        for (local_idx, (name, _)) in function_source_map.locals.iter().enumerate() {
               ...
        }
```
If we use `local_idx +parameter_len` to get the type, we will miss some of them as locals signature only contains locals types:
```
L0:	%#7: SomeEnum
L1:	%#8: u64
L2:	__match_tmp%#match_subject#2#2#0: u64
L3:	__match_tmp%#unpack_subject#1#1#0: ERROR[Unable to get type for local at index 6]
L4:	field#3#0: ERROR[Unable to get type for local at index 7]
L5:	res#1#0: ERROR[Unable to get type for local at index 8]
```

That being said, with the current solution locals indexes displayed with locals disassembly do not match what's in the disassembled bytecode due to what you observed ("in the bytecode all parameters are laid out in locals") and we need to bump them up using `parameter_len`:
```
public foo(e#0#0: SomeEnum, p1#0#0: u64, p2#0#0: u64): u64 * u64 {
L3:	%#7: SomeEnum
L4:	%#8: u64
L5:	__match_tmp%#match_subject#2#2#0: u64
L6:	__match_tmp%#unpack_subject#1#1#0: ERROR[Unable to get type for local at index 6]
L7:	field#3#0: ERROR[Unable to get type for local at index 7]
L8:	res#1#0: ERROR[Unable to get type for local at index 8]
B0:
	0: MoveLoc[0](e#0#0: SomeEnum)
	1: StLoc[6](__match_tmp%#unpack_subject#1#1#0: SomeEnum)
```